src/gluonnlp/adapters.py

```diff
@@ -0,0 +1,247 @@
+import math
+from collections import OrderedDict
+import mxnet as mx
+from mxnet import np, npx
+from mxnet import use_np
+from mxnet.gluon import nn, HybridBlock, Parameter, Constant
+import numpy as _np
+from .attention_cell import MultiHeadAttentionCell
+import mxnet.numpy_extension as _mx_npx
+from .layers import get_activation, get_norm_layer, _gen_repr_with_kwargs
+
+
+__all__ = ['BasicAdapter', 'AdapterFusion', 'AdapterModule', 'PositionwiseFFN_adapter']
+
+@use_np
+class BasicAdapter(nn.HybridBlock):
+    def __init__(self, config: dict, in_units: int):
+        super().__init__()
+        self._units = config['units']
+        self._activation = config['activation']
+        self.down_proj = nn.Dense(in_units=in_units,
+                                  units=self._units,
+                                  flatten=False,
+                                  weight_initializer=None, bias_initializer='zero')
+        self.activate = get_activation(self._activation)
+        self.up_proj = nn.Dense(in_units=self._units,
+                                  units=in_units,
+                                  flatten=False,
+                                  weight_initializer=None, bias_initializer='zero')
+
+
+
+    def forward(self, data, residual):
+        out = self.down_proj(data)
+        out = self.activate(out)
+        out = self.up_proj(out)
+        return out + residual

```

Comment:
You may not need a separate argument "`residual`" here. The residual connection refers to doing `return out + data`, where `data` is the original input before down projection, activation function and up projection.

![image](https://user-images.githubusercontent.com/946903/116631473-36024500-a923-11eb-8a16-34d366674779.png)
http://proceedings.mlr.press/v97/houlsby19a/houlsby19a.pdf